pkg/yang/file.go

```diff
@@ -115,22 +124,43 @@ func findFile(name string) (string, string, error) {
 	return "", "", fmt.Errorf("no such file: %s", name)
 }
 
-// findInDir looks for a file named name in dir or any of its subdirectories.
-func findInDir(dir, name string) string {
+// findInDir looks for a file named name in dir or any of its subdirectories if
+// recurse is true. if recurse is false, scan only the directory dir.
+func findInDir(dir, name string, recurse bool) string {
 	fis, err := ioutil.ReadDir(dir)
 	if err != nil {
 		return ""
 	}
+	var candidates []string
+	mname := name
+	if strings.HasSuffix(mname, ".yang") {
+		mname = name[:len(name)-len(".yang")]
+	}
+
 	for _, fi := range fis {
 		if !fi.IsDir() {
```

Comment:
This probably should be a switch statement:

switch {
case !fi.IsDir():
    ....
case recurse:
    ....
}

And then we can drop the continue